### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by DmitryHetman**

### CODE REVIEW

**Comment by DmitryHetman** [comment]
У меня все-равно сыпет ошибками при компиляции...
utils.c:102:17: warning: passing argument 2 of 'waddch' makes integer from pointer without a cast [-Wint-conversion]
                 mvaddch(t->me_y-1, t->me_x-1, t->skin);
                 ^
In file included from utils.h:16:0,
                 from utils.c:8:
/usr/include/curses.h:792:28: note: expected 'chtype {aka const unsigned int}' but argument is of type 'char _'
 extern NCURSES_EXPORT(int) waddch (WINDOW *, const chtype);  /_ implemented _/
                            ^
utils.c:103:17: warning: passing argument 2 of 'waddch' makes integer from pointer without a cast [-Wint-conversion]
                 mvaddch(t->me_y-1, t->me_x+1, t->skin);
                 ^
In file included from utils.h:16:0,
                 from utils.c:8:
/usr/include/curses.h:792:28: note: expected 'chtype {aka const unsigned int}' but argument is of type 'char *'
 extern NCURSES_EXPORT(int) waddch (WINDOW *, const chtype);  /_ implemented _/
                            ^
utils.c:109:17: warning: passing argument 2 of 'waddch' makes integer from pointer without a cast [-Wint-conversion]
                 mvaddch(t->me_y,   t->me_x, t->skin);
                 ^
In file included from utils.h:16:0,
                 from utils.c:8:
/usr/include/curses.h:792:28: note: expected 'chtype {aka const unsigned int}' but argument is of type 'char *'
 extern NCURSES_EXPORT(int) waddch (WINDOW *, const chtype);  /_ implemented _/
                            ^
utils.c:110:17: warning: passing argument 2 of 'waddch' makes integer from pointer without a cast [-Wint-conversion]
                 mvaddch(t->me_y-1, t->me_x, t->skin);
                 ^
In file included from utils.h:16:0,
                 from utils.c:8:
/usr/include/curses.h:792:28: note: expected 'chtype {aka const unsigned int}' but argument is of type 'char *'
 extern NCURSES_EXPORT(int) waddch (WINDOW *, const chtype);  /_ implemented _/
                            ^
utils.c:111:17: warning: passing argument 2 of 'waddch' makes integer from pointer without a cast [-Wint-conversion]
                 mvaddch(t->me_y+1, t->me_x, t->skin);
                 ^
In file included from utils.h:16:0,
                 from utils.c:8:
/usr/include/curses.h:792:28: note: expected 'chtype {aka const unsigned int}' but argument is of type 'char *'
 extern NCURSES_EXPORT(int) waddch (WINDOW *, const chtype);  /_ implemented _/
                            ^
utils.c:112:17: warning: passing argument 2 of 'waddch' makes integer from pointer without a cast [-Wint-conversion]
                 mvaddch(t->me_y,   t->me_x-1, t->skin);
                 ^
In file included from utils.h:16:0,
                 from utils.c:8:
/usr/include/curses.h:792:28: note: expected 'chtype {aka const unsigned int}' but argument is of type 'char *'
 extern NCURSES_EXPORT(int) waddch (WINDOW *, const chtype);  /_ implemented _/
                            ^
utils.c:113:17: warning: passing argument 2 of 'waddch' makes integer from pointer without a cast [-Wint-conversion]
                 mvaddch(t->me_y-1, t->me_x+1, t->skin);
                 ^
In file included from utils.h:16:0,
                 from utils.c:8:
/usr/include/curses.h:792:28: note: expected 'chtype {aka const unsigned int}' but argument is of type 'char *'
 extern NCURSES_EXPORT(int) waddch (WINDOW *, const chtype);  /_ implemented _/
                            ^
utils.c:114:17: warning: passing argument 2 of 'waddch' makes integer from pointer without a cast [-Wint-conversion]
                 mvaddch(t->me_y+1, t->me_x+1, t->skin);
                 ^
In file included from utils.h:16:0,
                 from utils.c:8:
/usr/include/curses.h:792:28: note: expected 'chtype {aka const unsigned int}' but argument is of type 'char *'
 extern NCURSES_EXPORT(int) waddch (WINDOW *, const chtype);  /_ implemented _/
                            ^
utils.c: In function 'boom_render':
utils.c:167:24: warning: implicit declaration of function 'abs' [-Wimplicit-function-declaration]
                     if(abs(t[k].me_x - b_bot[i].me_x) <= 1 && abs(t[k].me_y - b_bot[i].me_y) <=1 && t[k].live == 'Y' && b_bot[i].live == 'Y')
                        ^
cc -std=c99 -march=native -O2 -pipe   -c -o kernel.o kernel.c
gcc  main.o utils.o kernel.o -std=c99 -march=native -O2 -pipe  -lpthread -lncurses -o ttytanks
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: main.o: undefined reference to symbol 'halfdelay'
/lib64/libtinfo.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make: *_\* [Makefile:7: ttytanks] Error 1

**Comment by rainautumn** [comment]
тестил на FreeBSD и android только..
попробуй в Ubuntu 

```
sudo apt-get install libncurses5
```

Блин 5 утра! тебе делать нечего?

File: Makefile [hunk 1]
```diff
@@ -1,18 +1,15 @@
-CC = gcc -std=c99
```
Comment (DmitryHetman) [inline Makefile:1]: Не используестя же?

File: Makefile [hunk 2]
```diff
@@ -1,18 +1,15 @@
...
 OBJS = main.o utils.o kernel.o
-
-CFLAGS = -march=native -02 -pipe
+CFLAGS = -std=c99 -march=native -02 -pipe
 LDFLAGS =
```
Comment (DmitryHetman) [inline Makefile:3]: Почему тут пусто?

File: Makefile [hunk 3]
```diff
@@ -1,18 +1,15 @@
...
 
-LIBS =		-lpthread -lncurses
+LIBS = -lpthread -lncurses
 
-TARGET =	ttytanks
+TARGET = ttytanks
 
 $(TARGET):	$(OBJS)
-	g++  $(OBJS) $(CFLAGS) $(LDFLAGS) $(LIBS) -o $(TARGET)
```
Comment (DmitryHetman) [inline Makefile:12]: Игра же на си написана!

File: Makefile (hunk 2 above)
Comment (rainautumn) [inline Makefile:3]: что-то хотел написать, но видимо не захотел потом..

File: Makefile (hunk 3 above)
Comment (rainautumn) [inline Makefile:12]: на некоторых ос вылетает краши на линковке..

``````
                               /;\`                                           
                             `;//\/;/\```                                     
                             `^"";/\\\\;;;//\||`                              
                              `/;```"^:;\\//\\;/`                             
                              |::\|||;:""``^":;:`                             
                             `;;/`      `\:;:/|                               
                             //;`       `;;\                                  
                            \/:`       `///                                   
                           `;:|        \;;`                                   
                           ;;/        `;;`                                    
                          //;`       `;;\                                     
                         \/;`        ///                                      
                        `;;|        \/;`                                      
                        ;//        `;;`                                       
                       //;`       `;/\                                        
                      \/;`        ///                                         
                     `;;\`       \//                                          
                     ^^:\/;;\` `|;;`                                          
                    //`;\\|\\\\:^/\                                           
                   |/;|``|;:;/\^/``                                           
                   ;;|       `;/;`                                            
                  ///        |//`                                             
                 \/;         ;/|                                              
                `;;`        //\                                               
                ;/|        \\/                                                
               `//        |;/                                                 
               \/\       `/;                                                  
               //\      |//                                                   
               /:`     /\\                                                    
               :;    `//`                                                     
              |;;   /\/                                                       
              \/\ `;/|                                                        
             |/:||:/`                                                         
            |/^/;^/|                                                          
            /:/\^/;                                                           
           \/:|://                                                            
          |/^|/:/`                                                            
          /:\\:/|                                                             
         /;/\:/\                                                              
        \/;|://                                                               
       `/:|\:/                                                                
       /;\\:/|                                                                
      /;\|:/\                                                                 
     \\;|://                                                                  
    /;^|/;;                                                                   
    |/\|`\/                                                                   
    |\\/\                                                                     
   |\\"|                                                                      
  /\||/                                                                       
 |\\/::                                                                       
  `\\;        
``````

File: Makefile (hunk 1 above)
Comment (rainautumn) [inline Makefile:1]: используется при компиляции моего кода.
g++ только линкует

File: Makefile [hunk 4]
```diff
@@ -1,18 +1,15 @@
-CC = gcc -std=c99
 OBJS = main.o utils.o kernel.o
```
Comment (DmitryHetman) [inline Makefile:1]: CC = gcc

Comment (rainautumn) [inline Makefile:1]: я добавил гайд в readme для ubuntu и убрал лишние из Makefile 
СС = gcc по сути мало что в линуксе поменяет т.к. CC по дефолту равна gcc

**Comment by rainautumn** [comment]
приятной игры

**Pull request merged**